micro_mg3_0.F90

```diff
@@ -3389,22 +3644,26 @@ subroutine micro_mg_tend ( &
   ! calculate effective radius of rain and snow in microns for COSP using Eq. 9 of COSP v1.3 manual
 
   ! avoid divide by zero in avg_diameter_vec
+
+  !$acc loop gang vector collapse(2)
   do k=1,nlev
      do i=1,mgncol
         if (nrout(i,k) .eq. 0._r8) nrout(i,k)=1.e-34_r8
      end do
   end do
+  !$acc end parallel
 
   ! The avg_diameter_vec call does the actual calculation; other diameter
   ! outputs are just drout2 times constants.
-  ! call avg_diameter_vec(qrout,nrout,rho,rhow,drout2,mgncol*nlev)
+  call avg_diameter_vec(qrout,nrout,rho,rhow,drout2,mgncol*nlev)
```

Comment:
This is interesting. I adjusted this to maintain b4b as discussed in our emails. Knowing now that the previous avg_diameter function was elemental and changing every value as long as one met the conditions, I think that was a bug that you all fixed here. I'm glad this passes the ECT. I think we should leave it vectorized here and let this tag be answer changing.